File: sliding-window/slide-wind.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int findMax(int arr[], int lenarr, int k) {
    int maxVal = INT_MIN;
    int currentRunningSum = 0;

    int arrlength = lenarr;

    for (int i=0; i < arrlength; i++) {
        currentRunningSum += arr[i];
        cout << "i outside" << i << endl;
        if (i >= k - 1) {
            cout << "i inside" << i << endl;
            maxVal = std::max(maxVal, currentRunningSum);
            currentRunningSum -= arr[i - (k-1)];
        }
    }
    return maxVal;
}

int main(){
    int arr[10] = {4, 2, 1, 7, 8, 1, 2, 8, 1, 0}; // array data
    int k = 3; // jumlah urutan
    int lenarr = sizeof(arr)/sizeof(arr[0]);

    cout << findMax(arr, lenarr, k) << endl;


    return 0;
}

File: sliding-window/slide-wind.py
import sys

def find_max(arr, k):
    max_val = ~sys.maxsize
    current_running_sum = 0

    for i in range(len(arr)):
        current_running_sum += arr[i]
        if (i >= (k - 1)):
            max_val = max(max_val, current_running_sum)
            current_running_sum = current_running_sum - arr[i - (k-1)]

    return max_val


def main():
    arr = [4, 2, 1, 7, 8, 1, 2, 8, 1, 0]
    k = 3 # jumlah urutan

    print(find_max(arr, k))

if __name__ == "__main__":
    main()
